s3client.go

```diff
@@ -297,6 +298,9 @@ func (client *s3client) getBucketContents(bucketName string, prefix string) (map
 			} else {
 				marker = *listObjectsResponse.NextMarker
 			}
+			if marker == prevMarker {
```

Comment:
Technically it shouldn't happen since we don't use CloudFront for listing. However, I'd lean towards leaving it in as it manifested as a difficult to debug infinite loop.

Let us know if you want us to take it out — we will if that's the case.